docs/src/manual/constraints.md

```diff
@@ -55,4 +56,23 @@ apply!(K, f, ch)
 a = K\f
 apply!(a, ch) # enforces affine constraints
 
-```
\ No newline at end of file
+```
+
+Note: Replacing `f` with `r=-f` and solving `a = -K\r` in the example above may give the wrong result.
+Instead, use the procedure for nonlinear problems when applying affine constraints to such problem formulations.
```

Comment:
I don't really understand this part, and I don't think it is specific to affine constraints but inhomogeneities in general?

`apply!` essentially modifies `f` as: `f[free_dofs] = f[free_dofs] - K[free_dofs, constrained_dofs] * a[constrained]` where `a[constrained]` is the inhomogeneities. Note the `-` there. If you switch the sign of your right hand side, then that `-` should be a `+`. It feels a bit strange to modify the system your are solving with a `-` and then still using the same constraints (although I see why one would think this gives the same result).

Reply:
You're right! 
Perhaps this should rather be described in the doc-string of `apply!`?

Reply:
Yea, I think that would be better.